### PR TITLE
Add dictionary for pandas

### PR DESCRIPTION
**Pull request by aaronzipp**
pandas.txt covers all functions for pandas DataFrame, that aren't covered in other dictionaries.

### CODE REVIEW

**Comment by MichaelAquilina** [comment]
Thanks for this @aaronzipp. I've asked someone more familiar with pands to review this for me :)

**Comment by JelleZijlstra** [comment]
Perhaps a more general mechanism could be implemented that extracts all public methods from a package and uses them as an allowlist.

**Comment by MichaelAquilina** [comment]
> Perhaps a more general mechanism could be implemented that extracts all public methods from a package and uses them as an allowlist.

This has been attempted before but its not as easy as you would think with large, complex python projects :)

Feel free to give it a go though!

**Comment by JelleZijlstra** [comment]
I'm not volunteering :). An approach could be to use type stub files, e.g. https://github.com/VirtusLab/pandas-stubs, then use the AST to extract all function, class, and parameter names.

**Comment by aaronzipp** [comment]
> Perhaps a more general mechanism could be implemented that extracts all public methods from a package and uses them as an allowlist.

This is basically what I have done, when I have some time, I try to modify it so it can be done for arbitrary packages.

File: flake8_spellcheck/pandas.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+rsub
+rtruediv
+sem
+stata
+swapaxes
+swaplevel
+truediv
+tshift
+unstack
+xarray
```
Comment (MichaelAquilina) [inline flake8_spellcheck/pandas.txt:61]: It was suggested to me that these should also be added:

```
argmax
argmin
argsort
autocorr
backfill
barh
bfill
corr
endswith
extractall
ffill
fillna
findall
from_coo
fullmatch
hasnans
hist
idxmax
idxmin
isalnum
isalpha
isdecimal
isdigit
isin
islower
isna
isnull
isnumeric
isspace
istitle
isupper
ix
kde
kurt
ljust
lstrip
notnull
npoints
pydatetime
pytimedelta
qyear
ravel
reindex_like
rfind
searchsorted
skew
sp_values
startswith
std
strftime
swapcase
swaplevel
to_coo
to_csv
to_excel
to_hdf
to_markdown
to_sql
to_xarray
tz
unstack
var
zfill
```

what are your thoughts about adding these in?

Comment (aaronzipp) [reply flake8_spellcheck/pandas.txt:61]: The words with the underscore don't need to be included if the individual parts are in the file (e.g. for to_csv we have that "csv" is in the file and "to" is in the english dictionary). Besides that it looks good.
I actually worked on a way to automate the creation of a dictionary for abitrary packages and it gets almost everything, I just have to add another layer of recursion.
The only downside is that the package needs to be installed.

When it is working, should it be implemented such that the user can specify for which packages they want a dictionary  to be created?

Comment (MichaelAquilina) [reply flake8_spellcheck/pandas.txt:61]: > When it is working, should it be implemented such that the user can specify for which packages they want a dictionary to be created?

I'll probably need some thinking about where such a file could belong. This is tooling so it needs a bit of thought about how it gets installed.

Feel free to open a draft PR if you want to open a discussion though

**Comment by MichaelAquilina** [comment]
@aaronzipp merging this as is and I'll look into adding more at a later point.

Thanks for the PR :)

**Pull request merged**